Source/WebGPU/WGSL/TypeCheck.cpp

```diff
@@ -124,10 +126,55 @@ class UnificationContext {
     }
 
     SetForScope<UnificationContext*> m_setForScope;
-    Vector<std::pair<Type*, Type*>> m_constraints;
-    Vector<std::pair<Type*, SubstituionCallback>> m_substitutions;
+    WTF::Vector<std::pair<Type*, Type*>> m_constraints;
+    WTF::Vector<std::pair<Type*, SubstituionCallback>> m_substitutions;
 };
 
+TypeChecker::TypeChecker(AST::ShaderModule& shaderModule)
+    : m_shaderModule(shaderModule)
+    , m_typeConstrutors(AST::ParameterizedTypeName::NumberOfBaseTypes)
+{
+    m_abstractInt = allocateType<Primitive>(Primitive::AbstractInt);
+    m_abstractFloat = allocateType<Primitive>(Primitive::AbstractFloat);
+    m_void = allocateType<Primitive>(Primitive::Void);
+    m_bool = allocateType<Primitive>(Primitive::Bool);
+    m_i32 = allocateType<Primitive>(Primitive::I32);
+    m_u32 = allocateType<Primitive>(Primitive::U32);
+    m_f32 = allocateType<Primitive>(Primitive::F32);
```

Comment:
In WHLSL, we had these defined in a separate class (named `Intrinsics`) whose whole job was to implement the built-in types / functions. Should we do that too?

Reply:
That sounds good, I considered something similar but ended up not doing it. I'll land this and post another PR to refactor it.